### PR TITLE
FAT2-134 - Add Sort By Functionality to Filtered Searches

### PR DESCRIPTION
**Pull request by nouriach**
- Sort results by Name
- Sort results by Relevance
- Clear all sort/keyword filters
- Retain order and tiebreaker logic when Sectors and Levels are added to filter

### CODE REVIEW

**Comment by dashton82** [comment]
Empty feature file added that needs updating - also at some point will need to remove the Orderby.None and replace with null

File: src/SFA.DAS.FAT.Web/appsettings.json [hunk 1]
```diff
@@ -11,6 +11,6 @@
   },
   "FindApprenticeshipTrainingApi": {
     "Key": "test",
-    "BaseUrl": "http://localhost:5003/"
+    "BaseUrl": "https://localhost:5003/"
```
Comment (scottywakefield) [inline src/SFA.DAS.FAT.Web/appsettings.json:14]: let's keep these changes in `appsettings.Development.json`

Comment (dashton82) [reply src/SFA.DAS.FAT.Web/appsettings.json:14]: changed back

File: src/SFA.DAS.FAT.Web.UnitTests/Models/CoursesViewModelTests/WhenBuildingOrderByLinks.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            
+            // Act
+            var model = CoursesViewModelFactory.BuildModel(new List<Guid>(), null, selectedLevels);
+            
+            // Assert​
+            model.OrderByName.Should().Be($"?orderby={OrderBy.Name}&levels={string.Join("&levels=", selectedLevels)}");
+            model.OrderByRelevance.Should().Be($"?orderby={OrderBy.Relevance}&levels={string.Join("&levels=", selectedLevels)}");
+        }
+        [Test, AutoData]
+        public void And_Then_Adds_Sectors_And_Levels_To_Query_String(List<int> selectedLevels, List<Guid> selectedRoutes)
```
[the remark this answers is not in the document]
Comment (dashton82) [reply src/SFA.DAS.FAT.Web.UnitTests/Models/CoursesViewModelTests/WhenBuildingOrderByLinks.cs:57]: there is no more And_Then

File: src/SFA.DAS.FAT.Application/Courses/Queries/GetCourses/GetCoursesResult.cs [hunk 3]
```diff
@@ -10,5 +10,6 @@ public class GetCoursesResult
         public List<Level> Levels { get; set; }
         public int TotalFiltered { get ; set ; }
         public int Total { get ; set ; }
+        public OrderBy OrderBy { get; set; }
```
[the remark this answers is not in the document]
Comment (dashton82) [reply src/SFA.DAS.FAT.Application/Courses/Queries/GetCourses/GetCoursesResult.cs:13]: Removed - not used

File: src/SFA.DAS.FAT.Web.AcceptanceTests/Features/CourseDetail.feature [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿Feature: SpecFlowFeature1
+	In order to avoid silly mistakes
+	As a math idiot
```
[the remark this answers is not in the document]
Comment (dashton82) [reply src/SFA.DAS.FAT.Web.AcceptanceTests/Features/CourseDetail.feature:3]: very! removed

**Pull request merged**